cfme/tests/physical_infrastructure/api/test_physical_server_api.py

```diff
@@ -0,0 +1,87 @@
+# -*- coding: utf-8 -*-
+import pytest
+
+from cfme.physical.provider.lenovo import LenovoProvider
+from cfme.utils import error
+from cfme.utils.rest import assert_response
+from cfme.utils.wait import wait_for
+
+pytestmark = [
+    pytest.mark.tier(3),
+    pytest.mark.provider([LenovoProvider], scope='module')
+]
+
+# The max time that a single test has to perform the entire cycle of action
+# in seconds (5 min)
+TIMEOUT = 600
+
+# The time interval to perform a verification of state change
+# in seconds (1 min)
+DELAY = 60
+
+
+@pytest.fixture(scope="module")
+def physical_server(setup_provider_modscope, appliance):
+    physical_server = appliance.rest_api.collections.physical_servers[0]
+    return physical_server
+
+
+def get_server_attr(attr_name, provider, physical_server):
+    provider.refresh_provider_relationships()
+    physical_server.reload()
+    return physical_server[attr_name]
+
+
+def test_get_physical_server(physical_server, appliance):
+    existent_server = appliance.rest_api.get_entity('physical_servers', physical_server.id)
+    existent_server.reload
+    assert_response(appliance)
+
+
+def test_get_nonexistent_physical_server(appliance):
+    nonexistent = appliance.rest_api.get_entity('physical_servers', 999999)
+    with error.expected('ActiveRecord::RecordNotFound'):
+        nonexistent.reload()
+    assert_response(appliance, http_status=404)
+
+
+def test_invalid_action(physical_server, appliance):
+    payload = {
+        "action" : "invalid_action"
+    }
+    with error.expected('Api::BadRequestError'):
+        appliance.rest_api.post(physical_server.href, **payload)
+
+
+def test_refresh_physical_server(appliance, physical_server):
+    assert getattr(physical_server.action, "refresh")()
+    assert_response(appliance)
+
+
+@pytest.mark.parametrize("action, verification_attr, desired_state",[
```

Comment:
**REQUIRED:** Can we please add an ```ids``` parameter that just has the action name so that we don't have super long test ids. We can either just create a new list. Or do something like this

```python
actions = [
    ("power_off", "power_state", "off"),
    ("power_on", "power_state", "on"),
    ("power_off_now", "power_state", "off"),
    ("restart", "power_state", "on"),
    ("restart_now", "power_state", "on"),
    ("blink_loc_led", "location_led_state", "Blinking"),
    ("turn_on_loc_led", "location_led_state", "On"),
    ("turn_off_loc_led", "location_led_state", "Off")
]
@pytest.mark.parametrize("action, verification_attr, desired_state", actions, ids=[action[0] for action in actions])
......
```